pygmsh/geometry.py

```diff
@@ -344,13 +344,23 @@ def extrude(
             top = LineBase(top)
             extruded = SurfaceBase(extruded)
         elif isinstance(input_entity, SurfaceBase):
-            top = SurfaceBase(top)
+            top = SurfaceBase(input_entity.num_edges(), top)
             extruded = VolumeBase(extruded)
         else:
             top = Dummy(top)
             extruded = Dummy(extruded)
 
-        return top, extruded
+        lat = []
+        # lateral surfaces can be deduced only if we start from a SurfaceBase
```

Comment:
at line 287 I understand that `input_entity` types that are allowed are strings, `SurfaceBase`s or an object with the `surface` attribute, so `LineLoop`s are not allowed in this function.